app/uk/gov/hmrc/incorporatedentityidentificationfrontend/controllers/CheckYourAnswersController.scala

```diff
@@ -76,37 +76,50 @@ class CheckYourAnswersController @Inject()(journeyService: JourneyService,
+              case (LimitedCompany | RegisteredSociety, Some(companyProfile), Some(ctutr))  =>
+                validateIncorporatedEntityDetailsService.validateIncorporatedEntityDetails(companyProfile.companyNumber, ctutr)
+              case (RegisteredSociety, Some(companyProfile), None) => Future.successful(DetailsNotProvided)
+              case _ =>
+                throw new InternalServerException("No data stored")
+            }
+            _ <- details match {
+              case DetailsMatched =>
+                incorporatedEntityInformationService.storeIdentifiersMatch(journeyId, identifiersMatch = true)
+              case DetailsNotFound if isEnabled(EnableUnmatchedCtutrJourney) =>
+                for {
+                  _ <- incorporatedEntityInformationService.storeIdentifiersMatch(journeyId, identifiersMatch = false)
+                  _ <- incorporatedEntityInformationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
+                  _ <- incorporatedEntityInformationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+                } yield ()
+              case DetailsNotProvided => for {
+                _ <- incorporatedEntityInformationService.storeIdentifiersMatch(journeyId,identifiersMatch=false)
+                _ <- incorporatedEntityInformationService.storeBusinessVerificationStatus(journeyId,BusinessVerificationUnchallenged)
                 _ <- incorporatedEntityInformationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
               } yield ()
+              case _ =>
+                incorporatedEntityInformationService.storeIdentifiersMatch(journeyId, identifiersMatch = false)
+            }
+          } yield details match {
+            case DetailsMatched =>
+              Redirect(routes.BusinessVerificationController.startBusinessVerificationJourney(journeyId))
+            case DetailsNotFound if isEnabled(EnableUnmatchedCtutrJourney) =>
+              Redirect(routes.JourneyRedirectController.redirectToContinueUrl(journeyId))
+            case DetailsNotProvided =>
+              Redirect(routes.JourneyRedirectController.redirectToContinueUrl(journeyId))
             case _ =>
-              incorporatedEntityInformationService.storeIdentifiersMatch(journeyId, identifiersMatch = false)
+              Redirect(errorRoutes.CtutrMismatchController.show(journeyId))
           }
-        } yield details match {
-          case DetailsMatched =>
-            Redirect(routes.BusinessVerificationController.startBusinessVerificationJourney(journeyId))
-          case DetailsNotFound if isEnabled(EnableUnmatchedCtutrJourney) =>
-            Redirect(routes.JourneyRedirectController.redirectToContinueUrl(journeyId))
-          case _ =>
-            Redirect(errorRoutes.CtutrMismatchController.show(journeyId))
-        }
+        case None =>
+          throw new InternalServerException("Internal ID could not be retrieved from Auth")
```

Comment:
Instead of yielding the details and matching again, could we just return the redirect location within the for comprehension then yield that?